src/app/taskBoard/TaskBoard.tsx

```diff
@@ -104,25 +109,45 @@ export const TaskBoard = () => {
     [lists, moveCard],
   )
 
+  const { showModal } = useSelector(selectCreateTask)
+
   return (
-    <Stack direction="row" columnGap={2} sx={{ padding: '0px 20px', width: '100vw' }}>
-      {Object.entries(lists).map(([listId, listData], index) => (
-        <Droppable
-          key={listId}
-          accept={'taskCard'}
-          index={index}
-          id={Number(listId)}
-          onDropItem={(item) => onDropItem(Number(listId), item)}
+    <AppMargin size={SizeofAppMargin.LARGE} ptb="18.5px">
+      <Stack
+        direction="row"
+        columnGap={2}
+        sx={{
+          overflowX: 'auto',
+        }}
+      >
+        {Object.entries(lists).map(([listId, listData], index) => (
+          <Droppable
+            key={listId}
+            accept={'taskCard'}
+            index={index}
+            id={Number(listId)}
+            onDropItem={(item) => onDropItem(Number(listId), item)}
+          >
+            <TaskColumn key={listId} columnName={`List ${listId}`} taskCount={String(listData.length)}>
+              {listData.map((task: Task, index: number) => (
+                <Droppable key={task.id} accept={'taskCard'} index={index} id={Number(listId)} moveCard={moveCard}>
```

Comment:
@aatbip `Droppable` component having `moveCard` prop doesn't seem to match IMO, is it supposed to be something like `onDrop` or `onCardMove`

Reply:
`Droppable` component name has been changed to `DragDropHandler`. It is responsible for handling both drag and drop as necessary.